### PR TITLE
Fix issue 218 Conflicting import with Cats Implicits

### PR DESCRIPTION
**Pull request by zarthross**
Fixed the `diverging implicit expansion` error by using an Aux pattern on the `MaybeIsWritable` to hint the compiler  the correct F for implicit resolution

### CODE REVIEW

File: core/src/test/scala/ApiExamples.scala [hunk 1]
```diff
@@ -28,7 +29,7 @@ class ApiExamples extends Specification {
         val pathPart1 = GET / "hello"
 
         pathPart1 / "world" |>> { () => Ok("Hello, world!") }
-        pathPart1 / "you"   |>> { () => Ok("Hello, you!") }
+        pathPart1 / "you"   |>> { () => Ok("Hello, you!".pure[IO]) }
```
Comment (jcranky) [inline core/src/test/scala/ApiExamples.scala]: Why do we need this?

**Comment by bryce-anderson** [comment]
@zarthross, do you mean that you added so that we have a reason to import `cats.implicits._`? If so, I think that's fine, though I'd add that as code comment since it does look out of place.

**Comment by zarthross** [comment]
That is exactly why I added it.  I'll see if I can do something better/ more clear

**Comment by zarthross** [comment]
Added an explicit test, and move the `pure` in API examples to somewhere already using an `IO`.  Does this address you concerns?

File: core/src/test/scala/org/http4s/rho/bits/HListToFuncSpec.scala [hunk 2]
```diff
@@ -25,5 +25,14 @@ class HListToFuncSpec extends Specification {
       val req = Get("/route1")
       checkOk(req) should_== "foo"
     }
+
+    // Tests issue 218 https://github.com/http4s/rho/issues/218
+    "Work with cats.implicits" in {
+      import cats.implicits._
+      new RhoService[IO] {
+        GET / "route1" |>> { () => Ok("foo".pure[IO]) }
```
Comment (jcranky) [inline core/src/test/scala/org/http4s/rho/bits/HListToFuncSpec.scala]: Can we add a comment to say that the `pure[IO]` is here just for the import? My concern is people thinking that this is somewhat necessary.

Comment (zarthross) [reply core/src/test/scala/org/http4s/rho/bits/HListToFuncSpec.scala]: Done.

Comment (jcranky) [reply core/src/test/scala/org/http4s/rho/bits/HListToFuncSpec.scala]: thanks :D

**Pull request merged**